test/integration/node-specific/client_close.test.ts

```diff
@@ -0,0 +1,300 @@
+
+                const keyVaultClient = new MongoClient(uri);
+                await keyVaultClient.connect();
+                await keyVaultClient.db('keyvault').collection('datakeys');
+
+                const clientEncryption = new ClientEncryption(keyVaultClient, {
+                  keyVaultNamespace: 'keyvault.datakeys',
+                  kmsProviders
+                });
+                const dataKey = await clientEncryption.createDataKey(provider, { masterKey });
+
+                function getEncryptExtraOptions() {
+                  if (
+                    typeof process.env.CRYPT_SHARED_LIB_PATH === 'string' &&
+                    process.env.CRYPT_SHARED_LIB_PATH.length > 0
+                  ) {
+                    return { cryptSharedLibPath: process.env.CRYPT_SHARED_LIB_PATH };
+                  }
+                  return {};
+                }
+                const schemaMap = {
+                  'db.coll': {
+                    bsonType: 'object',
+                    encryptMetadata: {
+                      keyId: [dataKey]
+                    },
+                    properties: {
+                      a: {
+                        encrypt: {
+                          bsonType: 'int',
+                          algorithm: 'AEAD_AES_256_CBC_HMAC_SHA_512-Random',
+                          keyId: [dataKey]
+                        }
+                      }
+                    }
+                  }
+                };
+                const encryptionOptions = {
+                  autoEncryption: {
+                    keyVaultNamespace: 'keyvault.datakeys',
+                    kmsProviders,
+                    extraOptions: getEncryptExtraOptions(),
+                    schemaMap,
+                    tlsOptions: { aws: { tlsCAFile: infiniteFile } }
+                  }
+                };
+
+                const encryptedClient = new MongoClient(uri, encryptionOptions);
+                await encryptedClient.connect();
+
+                expect(process.getActiveResourcesInfo()).to.not.include('FSReqPromise');
+
+                const insertPromise = encryptedClient
+                  .db('db')
+                  .collection('coll')
+                  .insertOne({ a: 1 });
+
+                expect(process.getActiveResourcesInfo()).to.include('FSReqPromise');
+
+                await keyVaultClient.close();
+                await encryptedClient.close();
+
+                expect(process.getActiveResourcesInfo()).to.not.include('FSReqPromise');
+
+                const err = await insertPromise.catch(e => e);
+                expect(err).to.exist;
+                expect(err.errmsg).to.contain('Error in KMS response');
+              }
+            );
+          });
+        });
+      });
+
+      describe('Node.js resource: Socket', () => {
+        it('no sockets remain after client.close()', metadata, async () => {});
+      });
+    });
+  });
+
+  describe('Server resource: Cursor', () => {
```

Comment:
It may end up being a unit test but MongoClient close is going to iterate over a Set of "activeCursors" that it maintains and close them, so there's a client half to this that I just don't want us to forget. We can either leave a block here if it ends up making sense to use the tooling we have here or make a note in the cursors part of our tasks.

Reply:
resolving because we'll deal with this in the relevant ticket. There is no Node.js resource, just driver structures